File: search_algorithms/3_search_min_max/search_min_max.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        int[] arr = { 5, 3, 8, 2, 9, 1 };

        int minValue = FindMinimum(arr);
        int maxValue = FindMaximum(arr);

        Console.WriteLine($"Мінімальний елемент: {minValue}");
        Console.WriteLine($"Максимальний елемент: {maxValue}");
    }

    static int FindMinimum(int[] arr)
    {
        if (arr.Length == 0)
        {
            throw new ArgumentException("Масив порожній", nameof(arr));
        }

        int minVal = arr[0];
        foreach (int num in arr)
        {
            if (num < minVal)
            {
                minVal = num;
            }
        }
        return minVal;
    }

    static int FindMaximum(int[] arr)
    {
        if (arr.Length == 0)
        {
            throw new ArgumentException("Масив порожній", nameof(arr));
        }

        int maxVal = arr[0];
        foreach (int num in arr)
        {
            if (num > maxVal)
            {
                maxVal = num;
            }
        }
        return maxVal;
    }
}

File: algorithms/3 search_algorithms/2_search_min_max/search_min_max.py
def find_minimum(arr):
    if not arr:
        return None  # Повертає None, якщо масив порожній
    min_val = arr[0]  # Початкове значення мінімума
    for num in arr:
        if num < min_val:
            min_val = num  # Оновлюємо мінімальне значення, якщо знаходимо менше
    return min_val

def find_maximum(arr):
    if not arr:
        return None  # Повертає None, якщо масив порожній
    max_val = arr[0]  # Початкове значення максимума
    for num in arr:
        if num > max_val:
            max_val = num  # Оновлюємо максимальне значення, якщо знаходимо більше
    return max_val

# Приклад використання:
arr = [5, 3, 8, 2, 9, 1]
print("Мінімальний елемент:", find_minimum(arr))
print("Максимальний елемент:", find_maximum(arr))

